code/game/objects/items/bodybag.dm

```diff
@@ -328,11 +328,29 @@
 				var/stasis_ref = WEAKREF(H)
 				for(var/datum/data/record/R in GLOB.data_core.medical)
 					if (R.fields["ref"] == stasis_ref)
-						if(R.fields["last_scan_time"] && R.fields["last_scan_result"])
-							show_browser(usr, R.fields["last_scan_result"], "Last Medical Scan of [H]", "scanresults", "size=430x600")
+						if(R.fields["last_scan_time"] && R.fields["last_tgui_scan_result"])
+							tgui_interact(usr, human = H)
 						break
 
-
+/obj/structure/closet/bodybag/cryobag/tgui_interact(mob/user, datum/tgui/ui, var/mob/living/carbon/human/human)
+	. = ..()
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "HealthScan", "Last Medical Scan of [human]")
+		ui.open()
+		ui.set_autoupdate(FALSE)
+
+/obj/structure/closet/bodybag/cryobag/ui_static_data(mob/user)
+	if(ishuman(stasis_mob))
```

Comment:
static data is unchanging throughout the round so datacore data shouldnt be here since it changes